edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/disk/DiskSpaceChecker.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Storage.RocksDb.Disk
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage.Disk;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class DiskSpaceChecker : IStorageSpaceChecker
+    {
+        static readonly int checkFrequencyDefault = 3;
+        readonly string storageFolder;
+        readonly object updateLock = new object();
+        Option<DiskSpaceCheckerBase> inner;
+        long maxSize;
+        Option<int> checkFrequency;
+
+        DiskSpaceChecker(string storageFolder, Option<DiskSpaceCheckerBase> inner)
+        {
+            this.storageFolder = Preconditions.CheckNonWhiteSpace(storageFolder, nameof(storageFolder));
+            this.inner = Preconditions.CheckNotNull(inner, nameof(inner));
+        }
+
+        public bool IsFull => this.inner.Match<bool>(b => b.DiskStatus == DiskSpaceStatus.Full, () => false);
+
+        public static DiskSpaceChecker Create(string storageFolder)
+        {
+            // Start up diskSpaceChecker unfilled - so that it will be enabled when SetMaxSizeBytes is called
+            var diskSpaceChecker = new DiskSpaceChecker(storageFolder, Option.None<DiskSpaceCheckerBase>());
+            Events.Created(storageFolder);
+            return diskSpaceChecker;
+        }
+
+        public void SetMaxSizeBytes(long maxSizeBytes)
+        {
+            this.maxSize = maxSizeBytes;
+            this.EnableChecker();
+            Events.SetMaxSizeDiskSpaceUsage(this.maxSize, this.storageFolder);
+        }
+
+        public void SetCheckFrequency(Option<int> checkFrequencySecs)
+        {
+            checkFrequencySecs.ForEach(updatedCheckFrequency =>
+            {
+                if (updatedCheckFrequency > 0 && !this.checkFrequency.Equals(Option.Some(updatedCheckFrequency)))
+                {
+                    lock (this.updateLock)
+                    {
+                        this.checkFrequency = Option.Some(updatedCheckFrequency);
```

Comment:
>Option.Some(updatedCheckFrequency) [](start = 46, length = 34)

nit: You can pre-created this before the if condition and reuse it both in the if condition itself and here.